mule-deployer/src/main/java/org/mule/tools/client/standalone/deployment/StandaloneDeployer.java

```diff
@@ -0,0 +1,188 @@
+      throw new MojoFailureException("Error deploying application: [" + application + "]: " + e.getMessage());
+    } catch (RuntimeException e) {
+      throw new MojoExecutionException("Unexpected error deploying application: [" + application
+          + "]", e);
+    }
+  }
+
+  private void verifyMuleIsStarted() throws MuleControllerException {
+    log.debug("Checking if Mule Runtime is running.");
+    if (!mule.isRunning()) {
+      throw new MuleControllerException("Mule Runtime is not running! Aborting.");
+    }
+  }
+
+  private void waitForDeployments() throws MojoFailureException {
+    if (!application.exists()) {
+      throw new MojoFailureException("Application does not exists: " + application);
+    }
+    log.debug("Checking for application [" + application + "] to be deployed.");
+    String app = getApplicationName(application);
+    try {
+      prober.check(AppDeploymentProbe.isDeployed(mule, app));
+    } catch (AssertionError e) {
+      log.error("Couldn't deploy application [" + app + "] after [" + timeout
+          + "] miliseconds. Check Mule Runtime log");
+      throw new MojoFailureException("Application deployment timeout.");
+    }
+  }
+
+  private String getApplicationName(File application) {
+    String name = application.getName();
+    int extensionBeginning = name.lastIndexOf('.');
+    return extensionBeginning == -1 ? name : name.substring(0, extensionBeginning);
+  }
+
+  private void deployApplications() throws MojoFailureException {
+    log.info("Deploying application [" + application + "]");
+    try {
+      mule.deploy(application.getAbsolutePath());
+    } catch (MuleControllerException e) {
+      log.error("Couldn't deploy application: " + application + ". Check Mule Runtime logs");
+    }
+  }
+
+  private void startMuleIfStopped() {
+    log.debug("Checking if Mule Runtime is running.");
+    if (!mule.isRunning()) {
+      try {
+        log.info("Starting Mule Runtime");
+        if (arguments == null) {
+          mule.start();
+        } else {
+          mule.start(arguments);
+        }
+      } catch (MuleControllerException e) {
+        log.error("Couldn't start Mule Runtime. Check Mule Runtime logs.");
+      }
+    }
+  }
+
+  public StandaloneDeployer addLibraries(List<File> libs) {
+    for (File file : libs) {
+      mule.addLibrary(file);
+      log.debug(String.format("Adding library %s...", file));
+    }
+    return this;
+  }
+
+
+  public StandaloneDeployer addDomain(File domain) throws MojoFailureException {
+    try {
+      log.debug(String.format("Deploying domain : %s", domain));
+      mule.deployDomain(domain.getAbsolutePath());
+      return this;
+    } catch (MuleControllerException e) {
+      log.error("Couldn't deploy domain: " + domain);
+      throw new MojoFailureException("Couldn't deploy domain: " + domain);
+    }
+  }
+
+  public void addDependencies(DeploymentConfiguration configuration, ArtifactFactory factory, ArtifactResolver resolver,
```

Comment:
This guy should not deal with maven classes and artifact resolution. 
This specific stuff should be deal with at mojo level and just send a List<File> dependencies or just use the addLibraries